packages/devextreme/js/__internal/ui/chat/messagebox.ts

```diff
@@ -129,15 +150,54 @@ class MessageBox extends DOMComponent<MessageBox, Properties> {
   _createMessageSendAction(): void {
     this._messageSendAction = this._createActionByOption(
       'onMessageSend',
-      { excludeValidators: ['disabled', 'readOnly'] },
+      { excludeValidators: ['disabled'] },
+    );
+  }
+
+  _createTypingStartAction(): void {
+    this._typingStartAction = this._createActionByOption(
+      'onTypingStart',
+      { excludeValidators: ['disabled'] },
+    );
+  }
+
+  _createTypingEndAction(): void {
+    this._typingEndAction = this._createActionByOption(
+      'onTypingEnd',
+      { excludeValidators: ['disabled'] },
     );
   }
 
+  _triggerTypingStartAction(e: InputEvent): void {
+    if (!this._typingEndTimeoutId) {
+      this._typingStartAction?.({ event: e.event });
+    }
+  }
+
+  _updateTypingEndTimeout(): void {
+    clearTimeout(this._typingEndTimeoutId);
+
+    this._typingEndTimeoutId = setTimeout(() => {
+      this._typingEndAction?.();
+
+      this._typingEndTimeoutId = undefined;
+    }, TYPING_END_DELAY);
+  }
+
+  _clearTypingEndTimeout(): void {
+    clearTimeout(this._typingEndTimeoutId);
+
+    this._typingEndTimeoutId = undefined;
+  }
+
   _sendHandler(e: ClickEvent | EnterKeyEvent): void {
     if (!this._isValuableTextEntered()) {
       return;
     }
 
+    this._clearTypingEndTimeout();
```

Comment:
Should we do this when `this._typingEndTimeoutId` have value or should do it without any conditions? Kindly think about it

Reply:
There shouldn't be any conditions